### PR TITLE
P4-2514 adding links to navigate to map location.  Also fixing some of the back and cancel links to go back to journeys for review page.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Changes:

- Move summary page - 'Total moves by journey' renamed to 'Journeys for review'
- Wires in the navigation to the map location page from the journeys for review page.
- Displays the NOMIS agency ids instead of '**Not mapped**' in the journeys for review page.

@danielepolencic this addresses some of the issues Sangeetha has raised (I'll try and look at some more).

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt [hunk 1]
```diff
@@ -76,16 +82,23 @@ class HtmlController(@Autowired val moveService: MoveService, @Autowired val jou
...
             @ModelAttribute(name = DATE_ATTRIBUTE) @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) startOfMonth: LocalDate,
             @ModelAttribute(name = SUPPLIER_ATTRIBUTE) supplier: Supplier,
-            @ModelAttribute(name = "locationName") locationName: String,
-            @ModelAttribute(name = "agencyId") agencyId: String,
+            @ModelAttribute(name = "flashAttrMappedLocationName") locationName: String?,
+            @ModelAttribute(name = "flashAttrMappedAgencyId") agencyId: String?,
             model: ModelMap,
     ): String {
 
+        removeAttributesIf(locationName.isNullOrEmpty(), model, "flashAttrMappedLocationName", "flashAttrMappedAgencyId")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/HtmlController.kt:90]: The feels wrong but without this Thymeleaf was rendering text the first time you go into the journeys page when we don't want it.  It should only show the text if a location has been mapped.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/MapFriendlyLocationControllerTest.kt [hunk 2]
```diff
@@ -82,5 +84,25 @@ internal class MapFriendlyLocationControllerTest(@Autowired private val wac: Web
...
+  @Test
+  internal fun `map new location fails when duplicate location supplied supplied`() {
+    whenever(service.locationAlreadyExists(agencyId, "Duplicate location")).thenReturn(true)
+
+    mockMvc.post("/map-location") {
+      param("agencyId", agencyId)
+      param("locationName", "Duplicate location")
+      param("locationType", "CC")
+    }
+            .andExpect { model { attributeErrorCount("form", 1) } }
```
Comment (matt-crowson-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/controller/MapFriendlyLocationControllerTest.kt:101]: Not ideal I would rather check the error message.  Could do with looking at a later date...

File: src/main/resources/templates/map-location.html [hunk 3]
```diff
@@ -10,7 +10,7 @@
 <body>
 
 <div class="govuk-width-container" layout:fragment="content">
-    <a href="/dashboard" class="govuk-back-link">Back</a>
+    <a href="/journeys" class="govuk-back-link">Back</a>
```
Comment (matt-crowson-moj) [inline src/main/resources/templates/map-location.html:13]: This template serves as both update and create so this link may need to smarter in terms of where it goes...

File: src/main/resources/templates/map-location.html [hunk 4]
```diff
@@ -65,7 +65,7 @@ <h1 class="govuk-heading-xl mv2" th:inline="text">Update location</h1>
             <button class="govuk-button mv3 mr3" data-module="govuk-button">
                 Confirm and save
             </button>
-            <a href="/dashboard" class="govuk-link">Cancel</a>
+            <a href="/journeys" class="govuk-link">Cancel</a>
```
Comment (matt-crowson-moj) [inline src/main/resources/templates/map-location.html:68]: This template serves as both update and create so this link may need to smarter in terms of where it goes...

**Review by jcavell — APPROVED** [review]

**Pull request merged**